[CI] SearchableSnapshotsIntegTests testMountedSnapshotHasNoReplicasByDefault failing

Failed 7 times in the past week. Seems to only affect windows builds.

**Build scan:**
https://gradle-enterprise.elastic.co/s/xcsvhueyockt6/tests/:x-pack:plugin:searchable-snapshots:internalClusterTest/org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests/testMountedSnapshotHasNoReplicasByDefault

**Reproduction line:**
`gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests.testMountedSnapshotHasNoReplicasByDefault" -Dtests.seed=D78A0AE4D8EFBCB1 -Dtests.security.manager=true -Dtests.locale=es -Dtests.timezone=Asia/Beirut -Druntime.java=11`

**Applicable branches:**
7.x, master

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests&tests.test=testMountedSnapshotHasNoReplicasByDefault

**Failure excerpt:**
```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=1183, name=elasticsearch[node_s0][generic][T#6], state=RUNNABLE, group=TGRP-SearchableSnapshotsIntegTests]

  at __randomizedtesting.SeedInfo.seed([D78A0AE4D8EFBCB1:30058DB34F014821]:0)

  Caused by: java.lang.AssertionError: java.io.UncheckedIOException: java.nio.file.AccessDeniedException: C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2012-r2\x-pack\plugin\searchable-snapshots\build\testrun\internalClusterTest\temp\org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests_D78A0AE4D8EFBCB1-001\tempDir-002\node_s0\d0\indices\pqgpPEbCTTuwCx3-0yQzAg\6\snapshot_cache\39cQaFgeS467SQLd1kI40w\T6aoy2l4Qwy0LiABAWTv9A

    at __randomizedtesting.SeedInfo.seed([D78A0AE4D8EFBCB1]:0)
    at org.elasticsearch.common.util.concurrent.AbstractRefCounted.decRef(AbstractRefCounted.java:57)
    at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile.decrementRefCount(CacheFile.java:247)
    at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile.startEviction(CacheFile.java:275)
    at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:74)
    at org.elasticsearch.core.internal.io.IOUtils.closeWhileHandlingException(IOUtils.java:164)
    at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.onCacheFileEviction(CacheService.java:485)
    at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.lambda$new$1(CacheService.java:161)
    at org.elasticsearch.common.cache.Cache.delete(Cache.java:788)
    at org.elasticsearch.common.cache.Cache.lambda$new$6(Cache.java:488)
    at org.elasticsearch.common.cache.Cache$CacheSegment.remove(Cache.java:305)
    at org.elasticsearch.common.cache.Cache.invalidate(Cache.java:518)
    at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.processShardEviction(CacheService.java:409)
    at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService$1.doRun(CacheService.java:345)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
    at java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:669)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.lang.Thread.run(Thread.java:834)

    Caused by: java.io.UncheckedIOException: java.nio.file.AccessDeniedException: C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2012-r2\x-pack\plugin\searchable-snapshots\build\testrun\internalClusterTest\temp\org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests_D78A0AE4D8EFBCB1-001\tempDir-002\node_s0\d0\indices\pqgpPEbCTTuwCx3-0yQzAg\6\snapshot_cache\39cQaFgeS467SQLd1kI40w\T6aoy2l4Qwy0LiABAWTv9A

      at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile$1.closeInternal(CacheFile.java:71)
      at org.elasticsearch.common.util.concurrent.AbstractRefCounted.decRef(AbstractRefCounted.java:55)
      at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile.decrementRefCount(CacheFile.java:247)
      at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile.startEviction(CacheFile.java:275)
      at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:74)
      at org.elasticsearch.core.internal.io.IOUtils.closeWhileHandlingException(IOUtils.java:164)
      at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.onCacheFileEviction(CacheService.java:485)
      at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.lambda$new$1(CacheService.java:161)
      at org.elasticsearch.common.cache.Cache.delete(Cache.java:788)
      at org.elasticsearch.common.cache.Cache.lambda$new$6(Cache.java:488)
      at org.elasticsearch.common.cache.Cache$CacheSegment.remove(Cache.java:305)
      at org.elasticsearch.common.cache.Cache.invalidate(Cache.java:518)
      at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.processShardEviction(CacheService.java:409)
      at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService$1.doRun(CacheService.java:345)
      at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
      at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
      at java.util.concurrent.FutureTask.run(FutureTask.java:264)
      at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:669)
      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
      at java.lang.Thread.run(Thread.java:834)

      Caused by: java.nio.file.AccessDeniedException: C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2012-r2\x-pack\plugin\searchable-snapshots\build\testrun\internalClusterTest\temp\org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests_D78A0AE4D8EFBCB1-001\tempDir-002\node_s0\d0\indices\pqgpPEbCTTuwCx3-0yQzAg\6\snapshot_cache\39cQaFgeS467SQLd1kI40w\T6aoy2l4Qwy0LiABAWTv9A

        at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:89)
        at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
        at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
        at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:270)
        at sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
        at org.apache.lucene.mockfile.FilterFileSystemProvider.deleteIfExists(FilterFileSystemProvider.java:238)
        at java.nio.file.Files.deleteIfExists(Files.java:1180)
        at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile$1.closeInternal(CacheFile.java:69)
        at org.elasticsearch.common.util.concurrent.AbstractRefCounted.decRef(AbstractRefCounted.java:55)
        at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile.decrementRefCount(CacheFile.java:247)
        at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile.startEviction(CacheFile.java:275)
        at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:74)
        at org.elasticsearch.core.internal.io.IOUtils.closeWhileHandlingException(IOUtils.java:164)
        at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.onCacheFileEviction(CacheService.java:485)
        at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.lambda$new$1(CacheService.java:161)
        at org.elasticsearch.common.cache.Cache.delete(Cache.java:788)
        at org.elasticsearch.common.cache.Cache.lambda$new$6(Cache.java:488)
        at org.elasticsearch.common.cache.Cache$CacheSegment.remove(Cache.java:305)
        at org.elasticsearch.common.cache.Cache.invalidate(Cache.java:518)
        at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService.processShardEviction(CacheService.java:409)
        at org.elasticsearch.xpack.searchablesnapshots.cache.full.CacheService$1.doRun(CacheService.java:345)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:669)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.lang.Thread.run(Thread.java:834)

```

--- a/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/common/CacheFile.java
+++ b/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/common/CacheFile.java
@@ -6,6 +6,9 @@
  */
 package org.elasticsearch.xpack.searchablesnapshots.cache.common;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
 import org.apache.lucene.store.AlreadyClosedException;
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.action.support.PlainActionFuture;
@@ -17,7 +20,6 @@ import org.elasticsearch.common.util.concurrent.AbstractRunnable;
 import org.elasticsearch.core.internal.io.IOUtils;
 
 import java.io.IOException;
-import java.io.UncheckedIOException;
 import java.nio.channels.FileChannel;
 import java.nio.file.Files;
 import java.nio.file.Path;
@@ -34,6 +36,8 @@ import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Consumer;
 
 public class CacheFile {
+
+    private static final Logger logger = LogManager.getLogger(CacheFile.class);
 
     @FunctionalInterface
     public interface EvictionListener {
@@ -68,7 +72,8 @@ public class CacheFile {
             try {
                 Files.deleteIfExists(file);
             } catch (IOException e) {
-                throw new UncheckedIOException(e);
+                // nothing to do but log failures here since closeInternal could be called from anywhere and must not throw
+                logger.warn(() -> new ParameterizedMessage("Failed to delete [{}]", file), e);
             } finally {
                 listener.onCacheFileDelete(CacheFile.this);
             }
@@ -120,7 +125,8 @@ public class CacheFile {
             try {
                 fileChannel.close();
             } catch (IOException e) {
-                throw new UncheckedIOException(e);
+                // nothing to do but log failures here since closeInternal could be called from anywhere and must not throw
+                logger.warn(() -> new ParameterizedMessage("Failed to close [{}]", file), e);
             } finally {
                 decrementRefCount();
             }

